unit-tests/py/rspy/unittest.py

```diff
@@ -0,0 +1,349 @@
+                    del split_testname[1:i]
+                    found_test_dir = False
+                    break
+
+        path += os.sep + '-'.join( split_testname )
+        relative_path += '-'.join( split_testname )
+        if os.path.isfile( path + ".cpp" ):
+            relative_path += ".cpp"
+        elif os.path.isfile( path + ".py" ):
+            relative_path += ".py"
+        else:
+            log.w( log.red + self.name + log.reset + ':',
+                   'No matching .cpp or .py file was found; no configuration will be used!' )
+            return None
+
+        return relative_path
+
+
+class PyTest( Test ):
+    """
+    Class for python tests. Hold the path to the script of the test
+    """
+
+    def __init__( self, testname, path_to_test ):
+        """
+        :param testname: name of the test
+        :param path_to_test: the relative path from the current directory to the path
+        """
+        global unit_tests_dir
+        Test.__init__( self, testname )
+        self.path_to_script = unit_tests_dir + os.sep + path_to_test
+        self._config = TestConfigFromText( self.path_to_script, r'#\s*test:' )
+        relative_test_path = self.find_source_path()
+        if relative_test_path:
+            sub_dirs = relative_test_path.split( os.sep )[:-1] # last element will be the test's name
+            self._config.tags.update( sub_dirs )
+        self._config.tags.add( "py" )
+
+    def debug_dump( self ):
+        log.d( 'script:', self.path_to_script )
+        Test.debug_dump( self )
+
+    @property
+    def command( self ):
+        cmd = [sys.executable]
+        # The unit-tests should only find module we've specifically added -- but Python may have site packages
+        # that are automatically made available. We want to avoid those:
+        #     -S     : don't imply 'import site' on initialization
+        # NOTE: exit() is defined in site.py and works only if the site module is imported!
+        cmd += ['-S']
+        if sys.flags.verbose:
+            cmd += ["-v"]
+        cmd += [self.path_to_script]
+        if 'custom-args' not in self.config.flags:
+            if log.is_debug_on():
+                cmd += ['--debug']
+            if log.is_color_on():
+                cmd += ['--color']
+        return cmd
+
+    def run_test( self, configuration = None, log_path = None ):
+        try:
+            run( self.command, stdout=log_path, append=self.ran, timeout=self.config.timeout )
+        finally:
+            self._ran = True
+
+
+class ExeTest( Test ):
+    """
+    Class for c/cpp tests. Hold the path to the executable for the test
+    """
+
+    def __init__( self, testname, exe ):
+        """
+        :param testname: name of the test
+        :param exe: full path to executable
+        """
+        global unit_tests_dir
+        Test.__init__( self, testname )
+        self.exe = exe
+
+        relative_test_path = self.find_source_path()
```

Comment:
In the case of config, we have the relative path, right?
I suggest we pass it in in an optional argument to avoid failure-to-find...

Reply:
See my comment in config.py, first, though -- may not need to (or talk to me)